packages/ics721/src/types.rs

```diff
@@ -0,0 +1,68 @@
+use cosmwasm_schema::{cw_serde, schemars::JsonSchema};
+use cosmwasm_std::Binary;
+use serde::{Deserialize, Serialize};
+
+use crate::{ibc::NonFungibleTokenPacketData, token_types::ClassId};
+
+#[derive(Serialize, Deserialize, JsonSchema, Clone, Debug, PartialEq)]
+#[allow(clippy::derive_partial_eq_without_eq)]
+#[schemars(crate = "cosmwasm_schema::schemars")]
+#[serde(crate = "cosmwasm_schema::serde")]
+pub struct Ics721Memo {
+    pub callbacks: Option<Ics721Callbacks>,
```

Comment:
Also: why do we need an extra Ics721Callbacks prop?
Can't we just put both callbacks right away here in Ics721Memo?

Reply:
For it to be future proof.
Having this "extra" callback field will make it easier to add future features that relies on the memo, without breaking changes.
And will make it possible to handle 2 "memo features" in the same transfer.

Reply:
Got your point. Thx!